scenes/world_map/story_quest_starter.gd

```diff
@@ -0,0 +1,38 @@
+@tool
+class_name StoryQuestStarter
+extends Talker
+
+const STORY_QUEST_STARTER_DIALOGUE: DialogueResource = preload(
+	"res://scenes/world_map/story_quest_starter.dialogue"
+)
+
+## The first scene of a quest that this NPC offers to the player when they interact with them.
+@export var quest_scene: PackedScene
+
+## Dialogue line describing the quest; used by the default story_quest_starter dialogue
+@export var quest_description: String
+
+
+func _init() -> void:
+	# GDScript does not allow subclasses to override the default value of properties on the parent
+	# class. Fake this here – the default talker dialogue is certainly not wanted by instances of
+	# this class.
+	if dialogue == Talker.DEFAULT_DIALOGUE:
+		dialogue = STORY_QUEST_STARTER_DIALOGUE
+
+
+func _get_configuration_warnings() -> PackedStringArray:
+	if quest_scene:
+		return []
+
+	return ["Quest Scene property should be set"]
+
+
+## At the end of the current interaction, enter [member quest_scene]. This is intended to be called
+## from dialogue.
+func enter_quest() -> void:
+	interact_area.interaction_ended.connect(_do_enter_quest)
+
+
+func _do_enter_quest() -> void:
+	SceneSwitcher.change_to_packed(quest_scene)
```

Comment:
The reason it works like this is that, if you switch scenes while dialogue is running, the dialogue script will subsequently crash because its nodes have been removed from under it while it was `await`ing. I'm not wild about the API here and would welcome some other suggestions...

Reply:
I feel like it'd be clearer to follow the intention if the scene switch happened as part of `_on_dialogue_ended` 🤔 , what do you think?

Reply:
I'm trying to remember why I didn't do that.... I think it's because it means you need extra flag in the class for "should switch scene when the dialogue ends", and a super() call, so it ends up being more code, whereas the signal connection itself serves as that flag in the current implementation. (It should probably be a ONESHOT connection I guess.) I'm on the fence, I think you're probably right that the intention is clearer with what you propose.

Reply:
I pushed a patch that makes this change, adding a flag and overriding `_on_dialogue_ended()`. I'm fine with either implementation.